icechunk/Cargo.toml

```diff
@@ -55,6 +55,9 @@ tracing-subscriber = { version = "0.3.19", features = [
 tracing = "0.1.41"
 err-into = "1.0.1"
 serde_yaml_ng = "0.10.0"
+clap = { version = "4.5", features = ["derive"] }
+anyhow = "1.0.95"
+dialoguer = "0.11.0"
```

Comment:
What would be the impact of creating another library crate in the workspace and putting everything cli related there? Projects that want to depend on Icechunk as a Rust library, wouldn't need to bring in all these dependencies.